common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1527,64 +1527,57 @@ bool ap_wlan_hal_dwpal::read_acs_report()
 
 bool ap_wlan_hal_dwpal::read_supported_channels()
 {
-    char *reply = nullptr;
-    if (!dwpal_send_cmd("GET_HW_FEATURES", &reply)) {
-        LOG(ERROR) << __func__ << " failed";
-        return false;
-    }
+    auto ifname = get_radio_info().iface_name;
+    LOG(TRACE) << "for interface: " << ifname;
 
-    size_t replyLen = strnlen(reply, HOSTAPD_TO_DWPAL_MSG_LENGTH);
-    for (size_t i = 0; i < replyLen; i++) {
-        if (reply[i] == ')') {
-            reply[i] = ' ';
-        }
-    }
-
-    LOG(DEBUG) << "GET_HW_FEATURES replylen=" << (int)replyLen;
-    LOG(DEBUG) << "GET_HW_FEATURES reply=\n" << reply;
+    bwl::nl80211_client::radio_info radio_info;
 
-    size_t numOfValidArgs[2] = {0};
-    int chan                 = -1;
-    char dfs_state[50]       = {0};
-
-    FieldsToParse fieldsToParse[] = {
-        {(void *)&chan, &numOfValidArgs[0], DWPAL_INT_PARAM, "chan=", 0},
-        {(void *)dfs_state, &numOfValidArgs[1], DWPAL_STR_PARAM,
-         "(DFS state = ", sizeof(dfs_state)},
-        /* Must be at the end */
-        {NULL, NULL, DWPAL_NUM_OF_PARSING_TYPES, NULL, 0}};
-
-    rsize_t dmaxLen = (rsize_t)replyLen;
-    char *p2str     = nullptr;
-    char *lineMsg   = strtok_s(reply, &dmaxLen, "\n", &p2str);
-    int i           = 0;
-    while (lineMsg != NULL) {
-        size_t lineLen = strnlen(lineMsg, HOSTAPD_TO_DWPAL_MSG_LENGTH);
+    if (!m_nl80211_client->get_radio_info(ifname, radio_info)) {
+        LOG(TRACE) << "Failed to get channels info from nl80211";
+        return false;
+    }
+    std::vector<bwl::WiFiChannel> supported_channels;
+    for (auto const &band : radio_info.bands) {
+        for (auto const &pair : band.supported_channels) {
+            auto &channel_info = pair.second;
+            for (auto bw : channel_info.supported_bandwidths) {
+
+                bwl::WiFiChannel channel;
+                channel.channel = channel_info.number;
+                channel.tx_pow  = channel_info.tx_power;
+                channel.is_dfs  = channel_info.is_dfs;
+                switch (bw) {
```

Comment:
Please change this to use `beerocks::utils::convert_bandwidth_to_int()`

Reply:
changed